feat: Support WASM compilation and new JS interop

fuzzybinary opened the pull request:
### What and why?
This makes the necessary changes to support WASM compilation available in Flutter 3.22.

Using the new `js_interop` package requires use of Dart 3.3 (Flutter 3.19+), which is potentially a breaking change, so this needs to be discussed before it enters mainline.

refs: #615

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

fuzzybinary committed:
feat: Support WASM compilation and new JS interop

New `js_interop` package requires use of Dart 3.3, which is potentially a breaking change, so this needs to be discussed before it enters mainline.

refs: #615

0xnm commented on packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart:
```diff
@@ -135,5 +129,6 @@ class _JsUser {
 }
 
-@JS('setUser')
+// TODO: Need to fix this
```
should it be a part of this PR or it is good to go with TODO item?

fuzzybinary replied:
Nope, I should fix it. Thanks for calling it out.

fuzzybinary committed:
Fix issues with migration to package:web

fuzzybinary committed:
chore: Doc changes for WASM support

Add a changelog item and a README item specifying that 2.6+ will limit support to Flutter 3.19+

jhgilbert commented on packages/datadog_flutter_plugin/README.md:
```diff
@@ -4,6 +4,7 @@ Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real
 
 Datadog RUM SDK versions < 1.4 support monitoring for Flutter 2.8+.
 Datadog RUM SDK versions >= 1.4 support monitoring for Flutter 3.0+.
+Datadog RUM SDK versions >= 2.6 support monitoring for Flutter 3.19+
```
```suggestion
Datadog RUM SDK versions >= 2.6 support monitoring for Flutter 3.19+.
```

jhgilbert reviewed (dismissed):
Approved with a minor suggestion for consistency, thank you!

fuzzybinary committed:
Update packages/datadog_flutter_plugin/README.md

Co-authored-by: Jen Gilbert <j.h.gilbert@gmail.com>

0xnm commented on packages/datadog_flutter_plugin/integration_test_app/integration_test/rum_manual_error_reporting_test.dart:
```diff
@@ -56,7 +56,9 @@ void main() {
     expect(session.visits.length, 1);
 
     final view = session.visits[0];
+    print(view.viewEvents.last.view.errorCount);
     expect(view.viewEvents.last.view.errorCount, 3);
+    print(view.errorEvents.length);
```
some debugging leftovers?

fuzzybinary replied:
Yes, thanks. Nice catch.

fuzzybinary committed:
Remove unneeded debug prints
